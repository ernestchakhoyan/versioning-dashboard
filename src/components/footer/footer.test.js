import React from "react";
import { shallow } from "enzyme";
import {findBtTestAttr} from "../../utils"
import renderer from "react-test-renderer";
import Footer from "./index";


const setUp = (props = {}) => {
	return shallow(<Footer {...props} />);
};

describe("Footer component >", () => {

	let component;

	beforeEach(() => {
		component = setUp();
	});

	it("Should render without errors", () => {
		const footer = findBtTestAttr(component,"footer-component");
		expect(footer.length).toBe(1);
	});

	it("Should render logo", () => {
		const logo = findBtTestAttr(component,"footer-logo-img");
		expect(logo.length).toBe(1);
	});

	it("Capture component snapshot" ,() => {
		const tree = renderer.create(<Footer />).toJSON();

		expect(tree).toMatchSnapshot();
	})
});