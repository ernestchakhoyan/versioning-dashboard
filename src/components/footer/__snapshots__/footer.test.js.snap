// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component > Capture component snapshot 1`] = `
<footer
  className="footer"
  data-test="footer-component"
>
  <div
    className="footer__logo"
  >
    <img
      alt="logo"
      data-test="footer-logo-img"
      src="logo_portrait.png"
    />
  </div>
  <div
    className="footer__title"
  >
    Â© 2019 Behive, Inc. All Rights Reserved
  </div>
</footer>
`;
